File: back/src/main/java/org/example/control_user/service/RolService.java
package org.example.control_user.service;

import org.example.control_user.entity.Rol;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public interface RolService {
    Rol saveRol(Rol rol);

    List<Rol> getRols();

    Optional<Rol> getRol(Long id);

    void deleteRol(Long id);

    void updateRol(Rol rol);
}

File: front/src/services/RolService.js
import axios from 'axios';

const USER_API_BASE_URL = "http://89.117.109.48:8080/v1/rol/";

class RolService {

    getRols() {
        return axios.get(USER_API_BASE_URL);
    }

    createRol(rol) {
        return axios.post(USER_API_BASE_URL, rol);
    }

    getRolById(rolId) {
        return axios.get(USER_API_BASE_URL + rolId);
    }

    updateRol(rol) {
        return axios.put(USER_API_BASE_URL, rol);
    }

    deleteRol(rolId) {
        return axios.delete(USER_API_BASE_URL + rolId);
    }
}

export default new RolService()